### DISCUSSION TITLE
Obtaining voltage across an antenna, over a period of time

### DISCUSSION THREAD

**Discussion by Keith-A-Madison** [NanoComp/meep#2375] [Q&A]
I'm attempting to model a birdcage, bicone dipole VPol antenna, and a nearby VPol source. I'd like to simulate the voltage measured at the antenna as a function of time over some interval. 

I've managed to piece together a crude antenna model, and I'm wondering what the most "idiomatic" MEEP was is to model such an antenna. I see that material functions can be used to specify arbitrary shapes, however it isn't clear to me whether this is appropriate for my purposes.

Additionally, it isn't clear to me how one might most directly go about determining the voltage measured at the antenna (if this is possible).

**Comment by smartalecH** [comment]
In my experience, one typically uses a tool like FEM to get the near fields on a box around a complicated antenna geometry, and then imports these fields into an FDTD simulation and converts them into equivalent sources to see how the antenna interacts with a DUT. 

But if you want to analyze the time-domain response of the antenna itself, theoretically you could draw the antenna in meep using either geometric objects or a material function, like you suggest, and then _excite_ the antenna (this can be tricky to do properly... @stevengj might have some insight here). Since you're working with metals, the most "idiomatic" way to define your geometry is up to you. There's no subpixel smoothing benefits to be had from level-set geometry objects. So just do whatever is easiest. 

You can compute the voltage by spatially integrating the Electric field over a specified region. Doing this at multiple time steps (and multiple points) might be a bit tedious, and the results somewhat arbitrary unless you come up with a way to _normalize_ your fields by your excitation.
